openshift/oc-mirror
# CLID-53: Introduce env var to provide OCP release signature verificat…

# Description

Introduce env var to provide OCP release signature verification key

Fixes # [CLID-53](https://issues.redhat.com//browse/CLID-53)

:warning: **The commented unit test has shown that oc-mirror doesn't stop when signature is not verified**: Please advise if this is the requested behavior.

## Type of change

Please delete options that are not relevant.


# How Has This Been Tested?

- New unit tests have been introduced. 
- Execute oc-mirror mirror to disk with the following env var exported - logs should not show errors:
```
OCP_SIGNATURE_VERIFICATION_PK=./tests/custom-ocp-sig-key.asc
```
- Execute oc-mirror mirror to disk with the following env var exported - logs should show errors but not fail:
```
OCP_SIGNATURE_VERIFICATION_PK=./tests/buggy-ocp-sig-key.asc
```
```
2024/04/18 17:41:33  [ERROR]  : unexpected EOF
2024/04/18 17:42:25  [ERROR]  : invalid signature
2024/04/18 17:42:41  [ERROR]  : unexpected openpgp.MessageDetails: neither Signature nor SignatureV3 is set
```
